momento/storage_data_client.go

```
@@ -0,0 +1,130 @@
+	switch request.Value.(type) {
+	case StorageValueBytes:
+		val.Value = &pb.XStoreValue_BytesValue{BytesValue: request.Value.(StorageValueBytes)}
+	case StorageValueString:
+		val.Value = &pb.XStoreValue_StringValue{StringValue: string(request.Value.(StorageValueString))}
+	case StorageValueFloat64:
+		val.Value = &pb.XStoreValue_DoubleValue{DoubleValue: float64(request.Value.(StorageValueFloat64))}
+	case StorageValueInteger:
+		val.Value = &pb.XStoreValue_IntegerValue{IntegerValue: int64(request.Value.(StorageValueInteger))}
+	}
+
+	_, err := client.grpcClient.Put(requestMetadata, &pb.XStorePutRequest{
+		Key:   request.Key,
+		Value: &val,
+	})
+	if err != nil {
+		return nil, momentoerrors.ConvertSvcErr(err)
+	}
+
+	return &responses.StoragePutSuccess{}, nil
+}
+
+func (client *storageDataClient) get(ctx context.Context, request *StorageGetRequest) (responses.StorageGetResponse, momentoerrors.MomentoSvcErr) {
+	ctx, cancel := context.WithTimeout(ctx, client.requestTimeout)
+	defer cancel()
+
+	requestMetadata := metadata.NewOutgoingContext(
+		ctx, client.CreateNewMetadata(request.StoreName),
+	)
+
+	response, err := client.grpcClient.Get(requestMetadata, &pb.XStoreGetRequest{
+		Key: request.Key,
+	})
+
+	if err != nil {
+		return nil, momentoerrors.ConvertSvcErr(err)
+	}
+
+	val := response.GetValue()
+	switch val.Value.(type) {
+	case *pb.XStoreValue_BytesValue:
+		return responses.NewStoreGetSuccess_Bytes(responses.BYTES, val.GetBytesValue()), nil
+	case *pb.XStoreValue_StringValue:
+		return responses.NewStoreGetSuccess_String(responses.STRING, val.GetStringValue()), nil
+	case *pb.XStoreValue_DoubleValue:
+		return responses.NewStoreGetSuccess_Float64(responses.DOUBLE, val.GetDoubleValue()), nil
+	case *pb.XStoreValue_IntegerValue:
+		return responses.NewStoreGetSuccess_Integer(responses.INTEGER, int(val.GetIntegerValue())), nil
+	default:
+		return nil, momentoerrors.NewMomentoSvcErr(momentoerrors.UnknownServiceError, "Unknown store value type", nil)
```

slight deviation from what we'd aligned on, I think:
`Get` will return an error if the item is not found rather than returning a nil value

---

This is no longer true, right? As I understand it, the error returned on line 114 will prompt the storage client to return `nil, false, nil` for the `Get` call now?

---

Right, but it still won't compile without a `default` case there